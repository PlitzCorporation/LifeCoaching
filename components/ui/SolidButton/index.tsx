import RightArrow from '@/components/icons/RightArrow';
import Link from 'next/link';
import React from 'react';

interface SolidButtonProps {
	label: string;
	href?: string;
	onClick?: () => void;
}

const ButtonIcon = () => {
	return (
		<RightArrow className="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all" />
	);
};

const ButtonLabel = (props: { label: string }) => {
	const { label } = props;
	return (
		<span className="translate-x-2 group-hover:-translate-x-1 transition-all">
			{label}
		</span>
	);
};

const SolidButton = (props: SolidButtonProps) => {
	const { label, onClick, href } = props;

	const commonClasses =
		'relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl';

	return href ? (
		<Link href={href} className={commonClasses}>
			<ButtonLabel label={label} />
			<ButtonIcon />
		</Link>
	) : (
		<button onClick={onClick} className={commonClasses}>
			<ButtonLabel label={label} /> <ButtonIcon />
		</button>
	);
};

export default SolidButton;
