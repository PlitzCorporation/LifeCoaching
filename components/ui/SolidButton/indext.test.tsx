import { render } from '@testing-library/react';
import SolidButton from '.';

describe('SolidButton', () => {
	it('matches snapshot', () => {
		const { container } = render(<SolidButton label="Click Me" />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
