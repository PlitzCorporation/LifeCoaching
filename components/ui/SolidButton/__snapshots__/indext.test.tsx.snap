// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolidButton matches snapshot 1`] = `
<button
  class="relative flex items-center bg-plitz-primary text-white px-5 py-3 hover:rounded-xl border border-plitz-primary md:px-7 lg:px-9 md:py-4 hover:border-plitz-accent hover:text-plitz-accent hover:bg-white transition-all hover:transition-all group shadow-2xl"
>
  <span
    class="translate-x-2 group-hover:-translate-x-1 transition-all"
  >
    Click Me
  </span>
   
  <svg
    class="fill-white translate-x-0 opacity-0 group-hover:fill-plitz-accent group-hover:opacity-100 group-hover:translate-x-1 transition-all"
    height="1em"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M440.6 273.4c4.7-4.5 7.4-10.8 7.4-17.4s-2.7-12.8-7.4-17.4l-176-168c-9.6-9.2-24.8-8.8-33.9.8s-8.8 24.8.8 33.9L364.1 232H24c-13.3 0-24 10.7-24 24s10.7 24 24 24h340.1L231.4 406.6c-9.6 9.2-9.9 24.3-.8 33.9s24.3 9.9 33.9.8l176-168z"
    />
  </svg>
</button>
`;
